onnxruntime/core/providers/snpe/snpe_runtime_options.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/snpe/snpe_runtime_options.h"
+#include "core/common/logging/logging.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace snpe {
+
+void SnpeRuntimeOptions::ParseOptions() {
+  static const std::string OPT_RUNTIME = "runtime";
+  static const std::string OPT_PRIORITY = "priority";
+  static const std::string BUFFER_TYPE = "buffer_type";
+
+  // Option - Runtime
+  if (runtime_options_.find(OPT_RUNTIME) != runtime_options_.end()) {
+    runtime_target_ = SnpeRuntimeWrapper(runtime_options_[OPT_RUNTIME]);
+    LOGS_DEFAULT(INFO) << "Located user specified runtime target: " << runtime_options_[OPT_RUNTIME];
+  }
+  LOGS_DEFAULT(INFO) << "Runtime target: " << runtime_target_.ToString();
+
+  // Option Priority
+  if (runtime_options_.find(OPT_PRIORITY) != runtime_options_.end()) {
```

Comment:
general comment: avoid redundant lookups if possible, e.g. runtime_options_.find(OPT_PRIORITY) and then runtime_options_[OPT_PRIORITY].
alternatively:
```
if (const auto priority_opt_it = runtime_options_.find(OPT_PRIORITY);
    priority_opt_it != runtime_options_.end()) {
  if (priority_opt_it->second == "...")
  ...
}
```